front/src/components/boxs/device-in-room/device-features/ColorDeviceFeature.jsx

```diff
@@ -0,0 +1,126 @@
+import { Component, createRef, Fragment } from 'preact';
+import cx from 'classnames';
+import iro from '@jaames/iro';
+
+import { intToHexColor, hexToIntColor } from '../../../../utils/color';
+import { getDeviceName } from './utils';
+
+import style from './style.css';
+
+class ColorDeviceType extends Component {
+  colorPickerRef = createRef();
+
+  blur = event => {
+    if (!event.composedPath().includes(this.colorPickerRef.current.parentElement)) {
+      this.closeColorPicker(false);
+    }
+  };
+
+  scroll = () => {
+    this.colorPickerRef.current.parentElement.scrollIntoView({
+      behavior: 'smooth'
+    });
+  };
+
+  closeColorPicker = () => {
+    this.setColorPickerState(false);
+  };
+
+  toggleColorPicker = () => {
+    this.setColorPickerState(!this.state.open);
+  };
+
+  setColorPickerState = (open, fromEvent) => {
+    if (!open) {
+      document.removeEventListener('click', this.blur);
+    }
+
+    this.setState({ open, fromEvent }, () => {
+      if (this.state.open) {
+        this.scroll();
+        document.addEventListener('click', this.blur, true);
```

Comment:
Why don't you use React onClick?

Reply:
Can we use React onClick to detect a click outside a box?
This one is used to close the color popup when you click somewhere outside de color picker box.

Reply:
You're right it's probably the best way to do it! Read to fast, I didn't realized it was outside of the component :)